### ISSUE TITLE
serialize 0x prefix string object key as number

### ISSUE THREAD

**Issue by ryoid** [lxsmnsyc/seroval#38]
Encountered this while using SolidStart.

When object string key contains `0x` prefix it gets serialised as hex number. 

For context, address identifiers in crypto such as Ethereum are commonly hex addresses.

```tsx
import { deserialize, serialize } from "seroval"

const original = {
  "0x1": 2
}

const output = serialize(original)
console.log("output", output)
// output ({0x1:2})

const result = deserialize(output)
console.log("deserialize", result)
// deserialize { '1': 2 }
```

**Comment by lxsmnsyc** [comment]
Hmm I thought I handled this, will check

**Comment by lxsmnsyc** [comment]
should be fixed in 1.0.3
